lib/mix/tasks/compile.thrift.ex

```diff
@@ -70,8 +70,16 @@ defmodule Mix.Tasks.Compile.Thrift do
     Enum.empty?(targets) || Mix.Utils.stale?([thrift_file], targets)
   end
 
-  defp generate(thrift_file, output_dir, options) do
-    args = ~w[--gen erl --out] ++ [output_dir] ++ options ++ [thrift_file]
+  defp get_thrift_options(output_dir, user_options) do
+    opts = ~w[--out] ++ [output_dir]
+    # add --gen erl if user options doesn't contain --gen
+    opts = opts ++ if Enum.member?(user_options, "--gen"), do: ~w[--gen erl]
```

Comment:
Ah, no, it isn't.
My bad.
I should stop coding before the first cup of coffee.